tools/build-docs

```diff
@@ -18,9 +18,15 @@ from scripts.lib.zulip_tools import ENDC, WARNING, run
 def main() -> None:
     parser = argparse.ArgumentParser()
     parser.add_argument("-c", "--clean", action="store_true")
+    parser.add_argument("--watch", action="store_true")
```

Comment:
In trying to decide how and whether these two options should interact, I tried and failed to reproduce the claim in the warning below that “`tools/build-docs --clean` is necessary for the navigation/left sidebar to update”. This claim originated in #17961, which predates a major Sphinx upgrade, but I can’t reproduce it even with the version of the server and requirements from when that was reported (0267ba54b286a783dbde9a31d8ee2a0971671f73 with `s/clean html/html/`). Perhaps it’s a memory of a bug that was fixed even earlier, or perhaps I’m just trying the wrong thing.

@timabbott Do you have a way to reproduce this left sidebar update failure, or an upstream bug reference? If not, maybe we should remove the warning.

Reply:
Ah, I see the issue: you need `--clean` if you want edited headings on one page to show up in the sidebar *on other pages*.

You can get that in `sphinx-autobuild` with `-a`, but that’s slow since it rebuilds every page for every edit, so it probably doesn’t make sense to encourage that.